docs/FAQ.md

```diff
@@ -0,0 +1,124 @@
+
+
+### The compression process takes too long, how can I make it faster?
+For training approaches the majority of time is taken by the training loop, so any regular methods that improve model convergence should work here. 
+Try the built-in [knowledge distillation](./compression_algorithms/KnowledgeDistillation.md) to potentially obtain target accuracy faster.
+Alternatively you may want to reduce the number of initialization samples
+
+For post-training approaches... FILLME
+
+### I get a "CUDA out of memory" error when running NNCF in the compression-aware training approach, although the original model to be compressed runs and trains fine without NNCF.
+As some of the compression algorithm parameters are also trainable, NNCF-compressed model objects ready for training will have a larger GPU memory footprint than the uncompressed counterparts.
+Try reducing batch size for the NNCF training runs if it makes sense to do so in your situation.
+
+
+
+## PyTorch
+### Importing anything from `nncf.torch` hangs
+NNCF utilizes the [torch C++ extensions](https://pytorch.org/tutorials/advanced/cpp_extension.html) mechanism to accelerate the quantization-aware training process.
+This is done by just-in-time compiling a set of C++/CUDA files using the system-local compilers and toolsets.
+The compilation happens at the first import of `nncf.torch` or anything under that namespace on the machine, or within the current Python environment.
+The result is a set of the `.so` files containing the compiled extensions binary code stored in a system-specific location (commonly `~/.cache/torch_extensions`, or alternatively wherever `TORCH_EXTENSIONS_DIR` environment variable points to).
+
+To avoid race conditions, PyTorch uses `.lock` files in this folder during compilation.
+Sometimes, when the compilation process is abnormally aborted, these `.lock` files remain in the filesystem, which leads to a hang the next time you import anything from `nncf.torch`, because the just-in-time compilation process will wait indefinitely until the `.lock` files have been cleared.
+To resolve these, delete the `torch_extensions` directory (at `~/.cache`, or pointed to by `TORCH_EXTENSIONS_DIR`, or at your specific location), and re-run the script that imports from `nncf.torch`.
+The compilation takes some time and happens upon import, so do not interrupt the launch of your Python script until the import has been completed.
+
+### Importing anything from `nncf.torch` leads to an error mentioning `gcc`, `nvcc`, or `ninja`
+See the answer above for the general description of the reasons why these are involved in NNCF PyTorch operation.
+To resolve, make sure that your CUDA installation contains the development tools (e.g. the `nvcc` compiler), and that the environmental variables are set properly so that these tools are available in `PATH` or `PYTHONPATH`.
```

Comment:
On Windows you can see similar errors if you do not have a C++ compiler (Visual Studio) installed, or if the compiler is not in your PATH (it is a good idea to start NNCF from the developer prompt).

Reply:
Within NNCF, we don't really support Windows at the moment. However, I added the `cl.exe` reference in the header for something eye- and Google-catching so that folks that try NNCF on Windows ahead of time can still figure out what's wrong.